### ISSUE TITLE
💅 `noDeprecatedImports` reports an import when only one of its overloads is deprecated

### ISSUE THREAD

**Issue by AlbinoGeek** [biomejs/biome#11893]
### Environment

Biome 2.5.14 (CLI), Linux.

### Configuration

```json
{ "linter": { "rules": { "recommended": false, "suspicious": { "noDeprecatedImports": "error" } } } }
```

### Playground link / reproduction

`node_modules/lib/index.d.ts`:

```ts
export declare function make(url: string, options?: { modern?: true }): string;
/** @deprecated Use the modern options instead. */
export declare function make(url: string, options: { legacy: true }): string;
```

`src/index.ts`:

```ts
import { make } from 'lib'

// Resolves to the first overload, which is not deprecated.
export const value = make('https://example.test')
```

### Current behaviour

```
src/index.ts:1:10 lint/suspicious/noDeprecatedImports
  × Deprecated import: Use the modern options instead.
```

The import is reported as deprecated because one overload carries `@deprecated`, even though the call uses the other one.

### Expected behaviour

No diagnostic. An import should only be reported when the symbol itself is deprecated, or (ideally) when a call resolves to a deprecated overload.

### Real-world case

`@supabase/ssr` 0.12.7's `createBrowserClient` has two overloads; only the one taking the legacy `get`/`set`/`remove` cookie methods is `@deprecated`. `createBrowserClient(url, key)` (supabase's current recommended usage) is reported, so every project using it has to suppress the rule for that file.
